contracts/eternal-storage/EternalStorageProxy.sol

```diff
@@ -7,29 +7,29 @@ import "../interfaces/IEternalStorageProxy.sol";
 /**
  * @title EternalStorageProxy
  * @dev This proxy holds the storage of the token contract and delegates every call to the current implementation set.
- * Besides, it allows to upgrade the token's behaviour towards further implementations, and provides
+ * It allows to upgrade the token's behaviour towards further implementations, and provides
  * authorization control functionalities
  */
 contract EternalStorageProxy is EternalStorage, IEternalStorageProxy {
-    bytes32 internal constant OWNER = keccak256("owner");
 
     /**
-    * @dev This event will be emitted every time the implementation gets upgraded
-    * @param version representing the version number of the upgraded implementation
-    * @param implementation representing the address of the upgraded implementation
-    */
+     * @dev This event will be emitted every time the implementation gets upgraded
+     * @param version representing the version number of the upgraded implementation
+     * @param implementation representing the address of the upgraded implementation
+     */
     event Upgraded(uint256 version, address indexed implementation);
     
+    /**
+     * @dev This event will be emitted every time the ownership of this
+     * contract changes.
+     * @param previousOwner representing the previous owner of the contract
+     * @param newOwner representing the new owner of the contract
+     */
     event OwnershipTransferred(address indexed previousOwner, address indexed newOwner);
 
-    modifier onlyOwner() {
-        require(msg.sender == getOwner());
-        _;
-    }
-
     constructor(address _implementationAddress, address _owner) public {
         if (_implementationAddress != address(0)) {
-            require(_isContract(_implementationAddress));
+            require(_isContract(_implementationAddress) || 0 == block.number);
```

Comment:
`_isContract` doesn't work on genesis block?

Reply:
It does.  I thought that Parity initialized the contracts on the genesis block in unspecified order, so I thought the genesis block needed more special-casing than it actually does.